src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -93,10 +105,55 @@ public static int Run(string[] args)
 
                 msbuildArgs.AddRange(app.RemainingArguments);
 
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                ret._forwardingApp = new MSBuildForwardingApp(msbuildArgs, msbuildPath);
+
+                return 0;
             });
 
-            return app.Execute(args);
+            int exitCode = app.Execute(args);
+            if (ret._forwardingApp == null)
```

Comment:
when you pass invalid args and command line parser displays error message instead of executing